aplib.net-demo/Assets/Scripts/Player/Inventory.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Inventory : MonoBehaviour
+{
+    //TODO Add icon logic to Item class
+    //TODO Add hotbar Object which contains 4 subjobjects. They are linked to the inventory class and will display the right icons depending on inventory contents.
+    //TODO Add controls for using items to InputManager(?) class
+    //TODO Add controls for switching which item you're using to InputManager(?) class
+    //TODO Add logic for items with a duration and effect
+    Queue<Item> _itemList;
+    public float inventorySize;
+    public RawImage icon;
+    public Texture iconTexture;
+    public GameObject inventoryObject;
+    //here you would add all the possible items you can add, you add them to the inventory by calling PickUpItem method with the right item
+    public Item testItem;
+    public Item testItem2;
+    /// <summary>
+    /// Creates the inventory queue and sets default size, resets the items, and fetches the rawimage component to display the icons
+    /// </summary>
+    void Start()
+    {
+        //since we reuse items, it helps to reset the values for each of them on startup to avoid rare bugs with item uses
```

Comment:
This seems weird, a proper solution should be implemented, what bugs would even happen? Try to find the root cause and do not use hacky solutions.